recipes/libjpeg/all/conanfile.py

```diff
@@ -109,14 +128,33 @@ def _build_nmake(self):
                 replace_in_file(self, "Win32.Mak", "rc     = Rc", f"rc     = {rc}")
             # set flags directly in makefile.vc
             # cflags are critical for the library. ldflags and ldlibs are only for binaries
-            if is_msvc_static_runtime(self):
-                replace_in_file(self, "makefile.vc", "(cvars)", "(cvarsmt)")
-                replace_in_file(self, "makefile.vc", "(conlibs)", "(conlibsmt)")
+            if is_msvc(self):
+                msbuild = MSBuild(self)
+                if is_msvc_static_runtime(self):
+                    replace_in_file(self, "makefile.vs", "(cvars)", "(cvarsmt)")
+                    replace_in_file(self, "makefile.vs", "(conlibs)", "(conlibsmt)")
+                solution_version = 17 if msvs_toolset(self) == "v143" else 16
```

Comment:
if this is used for the `makejvcx.vXX`, these file names are fixed to the specific version of jpeg, as far as I can see - the filename has the `.v16` extension the most 3 recent releases of libjpeg